Sources/Core/AST/Stmt/ConditionalCompilationStmt.swift

```diff
@@ -94,10 +94,63 @@ public struct ConditionalCompilationStmt: Stmt {
 
   }
 
+  /// Abstracts the whole condition structure and its sub-conditions.
+  public indirect enum ConditionTree: Codable {
+
+    case operand(Condition)
+
+    case and(ConditionTree, ConditionTree)
+
+    case or(ConditionTree, ConditionTree)
+
+    public enum ConditionKind: String {
+      case holdOnly
+      case skipMain
+      case skipElse
+    }
+
+    /// Visit the SequenceCondition tail to calculate the right case for executing the right branch.
+    private func unroll(
+      for factors: ConditionalCompilationFactors, conditionCase: ConditionKind
+    ) -> Bool {
+      switch self {
+      case .and(let leftSubtree, let rightSubtree):
+        return leftSubtree.unroll(for: factors, conditionCase: conditionCase)
+          && rightSubtree.unroll(for: factors, conditionCase: conditionCase)
+
+      case .or(let leftSubtree, let rightSubtree):
+        return leftSubtree.unroll(for: factors, conditionCase: conditionCase)
+          || rightSubtree.unroll(for: factors, conditionCase: conditionCase)
+
+      case .operand(let cond):
+        switch conditionCase {
+        case .holdOnly: return cond.holds(for: factors)
+        case .skipMain: return cond.mayNotNeedParsing && !cond.holds(for: factors)
+        case .skipElse: return cond.mayNotNeedParsing && cond.holds(for: factors)
+        }
+      }
+    }
+
+    /// Return `true` iff the the full condition is satisfied.
```

Comment:
```suggestion
    /// Returns `true` iff the the condition is satisfied.
```
I don't think we ever care about a condition being partially satisfied.